axelrod/fingerprint.py

```diff
@@ -68,165 +66,17 @@ def __init__(self, strategy, probe=axl.TitForTat):
-        return probes
-
-    def construct_tournament_elements(self, step, progress_bar=True):
-        """Build the elements required for a spatial tournament
-
-        Parameters
-        ----------
-        step : float
-            The separation between each Point. Smaller steps will
-            produce more Points that will be closer together.
-        progress_bar : bool
-            Whether or not to create a progress bar which will be updated
-
-
-        Returns
-        ----------
-        edges : list of tuples
-            A list containing tuples of length 2. All tuples will have either 0
-            or 1 as the first element. The second element is the index of the
-            corresponding probe (+1 to allow for including the Strategy).
-
-        tournament_players : list
-            A list containing instances of axelrod.Player. The first item is the
-            original player, the rest are the probes.
-
-        """
-        self.points = create_points(step, progress_bar=progress_bar)
-        edges = self.create_edges(self.points, progress_bar=progress_bar)
-        probe_players = self.create_probes(self.probe, self.points,
-                                           progress_bar=progress_bar)
-
-        if isinstance(self.strategy, axl.Player):
-            tournament_players = [self.strategy] + probe_players
-        else:
-            tournament_players = [self.strategy()] + probe_players
-
-        return edges, tournament_players
-
-    @staticmethod
-    def generate_data(interactions, points, edges):
-        """Generates useful data from a spatial tournament.
-
-        Matches interactions from `results` to their corresponding Point in
-        `probe_points`.
-
-        Parameters
-        ----------
-        interactions : dictionary
-            A dictionary mapping edges to the corresponding interactions of
-            those players.
-        points : list
-            of Point objects with coordinates (x, y).
-        edges : list of tuples
-            A list containing tuples of length 2. All tuples will have either 0
-            or 1 as the first element. The second element is the index of the
-            corresponding probe (+1 to allow for including the Strategy).
-
-        Returns
-        ----------
-        point_scores : dictionary
-            A dictionary where the keys are Points of the form (x, y) and
-            the values are the mean score for the corresponding interactions.
-        """
-        edge_scores = [np.mean([
-            compute_final_score_per_turn(scores)[0]
-            for scores in interactions[edge]])
-            for edge in edges
-        ]
-        point_scores = dict(zip(points, edge_scores))
-        return point_scores
-
-    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None,
-                    filename=None, in_memory=False, progress_bar=True):
+        self.data = None  # type: PointsToFloats
```

Comment:
explicitly created all instance variable at __init__ and set to None with type hint.